### ISSUE TITLE
port 443 is not listening

### ISSUE THREAD

**Issue by mdsaifurrahmann** [andreapollastri/cipi#419]
hello, hope you mates are doing well.
I'm trying to use Cipi to my google cloud platform vm. but the problem is, port 443 is not listening on cipi. another vm instance with cyberpanel is just working fine with the same firewall configuration. but cipi is not listening to the port 443.
I don't want to touch or modify the nginx configuration file. though nginx also not listening to the port 443.
Can anyone lift any solution of this problem? I'm new user of cipi.

OS: Ubuntu 20.04 LTS.
RAM: 2GB
Storage: 20GB
UFW, Iptables installed and configured.
port 443 is opened by GCP firewall.

thanks in advance.

**Comment by Alnaggar94** [comment]
What you mean by not listening to port 443? I use cipi with google cloud without any issues.

After creating the site go to the site created then click on the generate SSL and wait for a few seconds it may take up to 5 minutes but usually within 10 seconds then try to access the site with https it will work normal

**Comment by mdsaifurrahmann** [comment]
> What you mean by not listening to port 443? I use cipi with google cloud without any issues.
> 
> After creating the site go to the site created then click on the generate SSL and wait for a few seconds it may take up to 5 minutes but usually within 10 seconds then try to access the site with https it will work normal

before modifying the nginx conf, It was refusing the connection while using https. but after adding this line `listen 443` to the nginx conf, now it's saying `saifur.xyz sent an invalid response.`
I tried to generate the ssl by clicking the generate SSL.
But it's not working.

**Comment by andreapollastri** [comment]
Hi @mdsaifurrahmann! I'm sorry for your issue but SSL works correctly within Cipi... I'm agree with @Alnaggar94! Are you behind any proxy? Did you correctly set you domain DNS? Try to run manually the certbot script and let us know the response!

**Comment by Alnaggar94** [comment]
@mdsaifurrahmann - 
I don't think hosting the DNS service is a good idea, it's better to do it with Cloudflare for many reasons such as your server goes down! and Cipi is different from Panels like cPanel and Plesk as those managing only one server, but Cipi allows you to manage multiple servers from the same panel 

@andreapollastri -
I'm really happy to see you back and can't wait for the v4, Are we going to see the Cloudflare integration with v4?

**Comment by mdsaifurrahmann** [comment]
@Alnaggar94 Yep, I know Cipi allows us to manage multiple server. I meant to creating addon-domain, sub domain without changing the DNS record from the Domain control panel. So I thought, It will reduce the time to configure a site to a developer. Literally I found potential to hosting DNS, as if I can control everything from Cipi. You know, sometimes DNS Propagation takes a lot. So...
But nothing serious. Just my opinion.
Thank you.

**Comment by Alnaggar94** [comment]
This is what the Cloudflare integration going to do like in Runcloud, Gridpane etc... Simply you add your Cloudflare info (Email,APi) then when you create a site it will automatically add a DNS record for the domain pointing to the right chosen server IP. and when you delete a site it will also delete the DNS record from Cloudflare. and also you will be able to manage DNS records from Cipi panel instead of the Cloudflare dashboard.
